xbmc/playlists/SmartPlayList.cpp

```diff
@@ -292,6 +296,63 @@ bool CSmartPlaylistRule::ValidateMyRating(const std::string &input, void *data)
   return StringValidation::IsPositiveInteger(input, data) && rating <= 10;
 }
 
+namespace
+{
+template<class T>
+std::optional<T> DigitsToNumeric(const std::string_view& str)
```

Comment:
The function body is fine as is, but please [pass `std::string_view` by value](https://quuxplusone.github.io/blog/2021/11/09/pass-string-view-by-value/).

As for the function: [`std::from_chars`](https://en.cppreference.com/w/cpp/utility/from_chars) could do all the validations for you but it's also fine as is.

Reply:
yes thanks, I just found out about from_chars and was rewritting.

Reply:
pushed an update, that's what you had in mind?

Reply:
Yes, exactly :slightly_smiling_face: